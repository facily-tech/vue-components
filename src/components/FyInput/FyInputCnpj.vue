<template>
  <div>
    <fy-input
      :value="text"
      @input="update"
      v-mask="mask"
      :rules="cnpjRule"
      v-bind.sync="$props"
      v-bind="$attrs"
      v-on="$listeners"
    />
  </div>
</template>

<script lang="ts">
import Vue from 'vue';

import { VTextField } from 'vuetify/lib';

const BaseVTextField = Vue.extend({ mixins: [VTextField] });

interface options extends InstanceType<typeof BaseVTextField> {
  value: string;
}

export default BaseVTextField.extend<options>().extend({
  name: 'fy-input-cnpj',

  props: {
    value: {
      type: String,
    },
  },

  data() {
    return {
      text: this.value ? this.value : '',
      cnpjRule: [
        (value: string): boolean | string => {
          if (value && value !== '')
            return value.length === 18 || 'O campo Cnpj deve conter 14 n√∫meros';

          return false;
        },
      ],
      mask: '##.###.###/####-##',
    };
  },

  methods: {
    update(value: string): void {
      this.$emit('input', value);
    },
  },
});
</script>
