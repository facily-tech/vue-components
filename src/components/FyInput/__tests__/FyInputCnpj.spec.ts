import Vue from 'vue';
import Vuetify from 'vuetify';
import FyInputCnpj from '../FyInputCnpj.vue';
import { mount, MountOptions, Wrapper } from '@vue/test-utils';

Vue.use(Vuetify);

describe('FyInputCnpj.ts', () => {
  // eslint-disable-line max-statements
  type Instance = InstanceType<typeof FyInputCnpj>;
  let mountFunction: (options?: MountOptions<Instance>) => Wrapper<Instance>;
  beforeEach(() => {
    mountFunction = (options?: MountOptions<Instance>) => {
      return mount(FyInputCnpj, {
        vuetify: new Vuetify(),
        ...options,
      });
    };
  });

  it('should render component and match snapshot', () => {
    const wrapper = mountFunction();

    expect(wrapper.html()).toMatchSnapshot();
  });
});
