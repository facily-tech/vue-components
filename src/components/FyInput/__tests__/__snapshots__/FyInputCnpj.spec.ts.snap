// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FyInputCnpj.ts should render component and match snapshot 1`] = `
<div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined" value="" rules="(value) => {
          /* istanbul ignore next */
          cov_22qvzt4d0l.f[2]++;
          cov_22qvzt4d0l.s[9]++;

          if (
          /* istanbul ignore next */
          (cov_22qvzt4d0l.b[2][0]++, value) &amp;&amp;
          /* istanbul ignore next */
          (cov_22qvzt4d0l.b[2][1]++, value !== '')) {
            /* istanbul ignore next */
            cov_22qvzt4d0l.b[1][0]++;
            cov_22qvzt4d0l.s[10]++;
            return (
              /* istanbul ignore next */
              (cov_22qvzt4d0l.b[3][0]++, value.length === 18) ||
              /* istanbul ignore next */
              (cov_22qvzt4d0l.b[3][1]++, 'O campo Cnpj deve conter 14 números')
            );
          } else
          /* istanbul ignore next */
          {
            cov_22qvzt4d0l.b[1][1]++;
          }

          cov_22qvzt4d0l.s[11]++;
          return false;
        }">
  <div class="v-input__control">
    <div class="v-input__slot">
      <fieldset aria-hidden="true">
        <legend style="width: 0px;"><span class="notranslate">​</span></legend>
      </fieldset>
      <div class="v-text-field__slot"><input data-test="input" id="input-3" type="text"></div>
    </div>
    <div class="v-text-field__details">
      <div class="v-messages theme--light">
        <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
      </div>
    </div>
  </div>
</div>
`;
